geetools/ee_image_collection.py

```diff
@@ -1740,3 +1740,129 @@ def to_int(d):
+            propertyReducer: If the multiple images have the same property, they will be aggregated beforehand using the provided reducer. default to a mosaic behaviour to match most of the satellite imagery collection where the world is split for each date between multiple images.
+            dateFormat: If a date format is used for the IdProperty, the values will be formatted as "YYYY-MM-ddThh-mm-ss". You can specify any other format compatible with band names.
+            numberFormat: If a number format is used for the IdProperty, the values will be formatted as a string  ("%s"). You can specify any other format compatible with band names.
+            geometry: The region over which to reduce the data.
+            scale: A nominal scale in meters to work in.
+            crs: The projection to work in. If unspecified, the projection of the image's first band is used. If specified in addition to scale, rescaled to the specified scale.
+            crstransform: The list of CRS transform values. This is a row-major ordering of the 3x2 transform matrix. This option is mutually exclusive with 'scale', and replaces any transform already set on the projection.
+            bestEffort: If the polygon would contain too many pixels at the given scale, compute and use a larger scale which would allow the operation to succeed.
+            maxPixels: The maximum number of pixels to reduce.
+            tileScale: A scaling factor between 0.1 and 16 used to adjust aggregation tile size; setting a larger tileScale (e.g., 2 or 4) uses smaller tiles and may enable computations that run out of memory with the default.
+
+        Returns:
+            A dictionary with the reduced values for each image.
+
+        Examples:
+            .. code-block:: python
+
+                import ee, geetools
+
+                ee.Initialize()
+
+                collection = (
+                    ee.ImageCollection("LANDSAT/LC08/C01/T1_TOA")
+                    .filterBounds(ee.Geometry.Point(-122.262, 37.8719))
+                    .filterDate("2014-01-01", "2014-12-31")
+                )
+                data = collection.geetools.reduceRegion("mean", geometry=ee.Geometry.Point(-122.262, 37.8719), scale=30)
+                print(data.getInfo())
+        """
+        # filter the imageCollection with the region parameter to reduce the number of manipulated images and speed up the computation
+        ic = self._obj.filterBounds(geometry)
+
+        # raise an error if the idPropertyType is not supported
+        if idPropertyType not in [ee.String, ee.Number, ee.Date]:
+            msg = f"idPropertyType format {idPropertyType} not supported (yet)!"
+            raise ValueError(msg)
+
+        # create a unique property name to avoid conflict with any
+        # existing property in the image collection
+        pname = uuid.uuid4().hex
```

Comment:
this will block the possibility to use this method on the server-side. However, if there will be a reduceRegions method to apply the same over a set of features (ee.FeatureCollection) then it should be ok.

Reply:
another possibility would be to create something like `_geetools_ic_reduce_region_` which will have probably a similar probabilities of existence than a uuid4 hex, but would allow it to run on the server-side. This is exactly what I proposed in [this branch](https://github.com/gee-community/geetools/tree/reduce-ic-same-property).

Reply:
I don't understand what you mean with "server-side". I'm simply creating a fake name, the whole computation is going to be perform server side. at the end of the day, the `pname` value act as any other string. If what you mean is the javascript interface, I decided to stop caring a long time ago as they will never support ECMA6 and thus prevent any extention mechanism.

Reply:
Ahhhhh ok I get it, you mean it will not be possible to use this function in a server-side `map` right ? 
I think it's more desirable to have a proper `reduceRegions` instead.

Reply:
Exactly that. I agree that having a `reduceRegions` is the way to go, since happens the same with the original functions; when you want to reduce many features you call `reduceRegions`. However, I don't see the issue on using a fix string, which could contain 32 chars as uuid. Without any statistical proof, I think it has a similar (or even the same) probability of collision. I asked Copilot and Gemini and the answers were interesting. The first insisted on uuid due to "avoiding collisions" and "consistency" while the latter went for the fix string due to "generation simplicity", "performance" and "storage efficiency" and considering "uniqueness" is not a key factor in this use case.

Let's go for uuid and write in the docstring that it cannot be used in a server-side function (e.g. in a `map` operation) and point in the right direction (to use `reduceRegions` instead).

Reply:
solved for now in https://github.com/gee-community/geetools/pull/359/commits/5e8d4e129a90eef10bc750e9bfa24414b892c29a